x-pack/plugins/enterprise_search/common/__mocks__/flush_promises.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+/*
+ * This function ends the current event loop which will give all promises a chance
+ * to complete. It is similar to calling `setTimeout` without specifying a length of
+ * time.
+ *
+ * For example:
+ *   // Mock http.get to return a rejected promise immediately
+ *   (HttpLogic.values.http.get as jest.Mock).mockReturnValue(Promise.reject('An error occured'));
+ *   // Call an action, which calls http.get and awaits it
+ *   CredentialsLogic.actions.fetchCredentials(2);
+ *   // Give the http.get promise a chance to be rejected
+ *   await flushPromises();
+ *   // Make assertions
+ *   expect(flashAPIErrors).toHaveBeenCalledWith('An error occured');
+ */
+export const flushPromises = () => {
```

Comment:
I added this to help with Jest tests:

Initial implementation: 77e519bf7c24adf77e4d69e2fadff76293ba810e
Updated to arrow syntax: a0d519ecb8aa9959cace4395bd2c8fe97d4e5b66

Reply:
(Repeating our Slack convo here to close the loop/capture context in GitHub up to date): it looks like there's an alternate solution we can use for now where we [store and await our nested promises](https://stackoverflow.com/questions/50248925/jest-testing-a-nested-promise). It'd be great to switch to that for now in this PR - we might need flushPromises down the road, but hopefully not just yet? 🤞

Reply:
I struggled for a bit with the `Promise.reject` statement, which now fails the jest test because it is an unhandled promise rejection. I ended up just needing to add a try / catch to the `await` though: https://github.com/elastic/kibana/pull/77626/commits/585f6e664dbef19ad73dd01eac6ee0bd560deb6f

```ts
      const promise = Promise.reject('An error occurred');
      (HttpLogic.values.http.get as jest.Mock).mockReturnValue(promise);

      CredentialsLogic.actions.fetchCredentials();

      try {
        await promise;
      } catch {
        // Ignore this
      }
```